### PR TITLE
Sign in errors

### PR DESCRIPTION
**Pull request by cheddartot**

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -70,13 +70,16 @@ def signup():
     form_sign_up = SignInForm()
     if current_user.is_authenticated:
         return redirect('/')
+    if not(utils.find_user_by_username(form_sign_up.username.data) is None):
```
Comment (pgayee694) [inline app/routes.py]: simplify this: in python you can do null checks by just doing "if object"

File: app/routes.py [hunk 2]
```diff
@@ -70,13 +70,16 @@ def signup():
...
+        return render_template('signup.html', title='Sign Up', form=form_sign_up, user_exists=True)
+    if not(form_sign_up.password.data == form_sign_up.password2.data):
```
Comment (pgayee694) [inline app/routes.py]: simplify

**Review by bbuffington1662 — APPROVED** [review]

**Review by pgayee694 — CHANGES_REQUESTED** [review]

**Review by tkorisko — APPROVED** [review]
👌👌

File: app/routes.py [hunk 3]
```diff
@@ -70,13 +70,16 @@ def signup():
     form_sign_up = SignInForm()
     if current_user.is_authenticated:
         return redirect('/')
+    if utils.find_user_by_username(form_sign_up.username.data):
+        return render_template('signup.html', title='Sign Up', form=form_sign_up, user_exists=True)
+    if form_sign_up.password.data != form_sign_up.password2.data:
```
Comment (pgayee694) [inline app/routes.py:87]: put a line of space between different if blocks

File: app/routes.py [hunk 4]
```diff
@@ -57,10 +57,10 @@ def login():
     if form_log_in.validate_on_submit():
         flash('Login requested for user {}, remember_me={}'.format(
             form_log_in.username.data, form_log_in.remember_me.data))
-        user = models.User.query.filter_by(username=form_log_in.username.data).first()
+        user = utils.find_user_by_username(form_log_in.username.data)
         if user is None or not user.check_password(form_log_in.password.data):
             flash('invalid username or password')
-            return redirect('/login/')
+            return render_template('login.html', title='Log In', form=form_log_in, invalid=True)
```
Comment (pgayee694) [inline app/routes.py]: you should display error messages to the user like this via flash(). Here's a good example: https://stackoverflow.com/questions/40949746/how-to-display-flashing-message-without-reloading-the-page-in-flask

**Review by pgayee694 — CHANGES_REQUESTED** [review]

**Review by pgayee694 — APPROVED** [review]

**Pull request merged**